### PR TITLE
Feature/bimpm

### PR DESCRIPTION
**Pull request by rgtjf**
#183 Not done yet.

This PR is from https://github.com/NTMC-Community/MatchZoo/pull/589

### CODE REVIEW

File: matchzoo/contrib/models/bimpm_model.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+from keras.layers import Dense, Concatenate, Dropout
+from keras.layers import Bidirectional, LSTM
+
+from matchzoo.engine.param import Param
+from matchzoo.engine.param_table import ParamTable
+from matchzoo.engine.base_model import BaseModel
+from matchzoo.contrib.layers import MultiPerspectiveLayer
+
+
+class BimpmModel(BaseModel):
```
Comment (faneshion) [inline matchzoo/contrib/models/bimpm_model.py]: Please rename the model name as `BiMPM`.

File: matchzoo/contrib/models/bimpm_model.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    """
+    BimpmModel.
```
Comment (faneshion) [inline matchzoo/contrib/models/bimpm_model.py]: BiMPM.

File: matchzoo/contrib/models/bimpm_model.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+"""Bimpm."""
```
Comment (faneshion) [inline matchzoo/contrib/models/bimpm_model.py]: BiMPM

File: matchzoo/contrib/models/__init__.py [hunk 4]
```diff
@@ -1 +1,2 @@
 from .match_lstm import MatchLSTM
+from .bimpm_model import BimpmModel
```
Comment (faneshion) [inline matchzoo/contrib/models/__init__.py]: BiMPM

**Comment by rgtjf** [comment]
`make flake` ignores `,matchzoo/contrib`.

**Comment by rgtjf** [comment]
line 394 MpCosineLayer
>>> layer.compute_output_shape([(32, 10, 100), (32, 40, 100)])
        (32, 10, 40, 50)
there is problem in output value. 
hold on.

**Pull request merged**